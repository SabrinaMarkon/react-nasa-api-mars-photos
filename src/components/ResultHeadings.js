import React, { Component } from 'react';

export default class ResultHeadings extends Component {

  render() {
    return (
      <>
      <div className="grid-item grid-heading">Name</div>
      <div className="grid-item grid-heading">Id</div>
      <div className="grid-item grid-heading">Name Type</div>
      <div className="grid-item grid-heading">Rec Class</div>
      <div className="grid-item grid-heading">Mass (g)</div>
      <div className="grid-item grid-heading">Fall</div>
      <div className="grid-item grid-heading">Year</div>
      <div className="grid-item grid-heading">Latitude</div>
      <div className="grid-item grid-heading">Longitude</div>
      </>
    );
  }
}