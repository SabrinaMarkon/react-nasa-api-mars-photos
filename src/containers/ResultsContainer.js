import React, { Component, Fragment } from 'react';
import ResultHeadings from '../components/ResultHeadings';

export default class ResultsContainer extends Component {

  render() {
    let searchResults = Array.from(this.props.searchResults);

    // format the year property to only show 4 digit year.
    searchResults.forEach(result => {
      if (result.year) {
        result.year = result.year.slice(0, 4);
      }
    });

    let results = searchResults.map((row,i) => {
      return (
        <Fragment key={i}>
          <div className="grid-item" key={i}>{row.name}</div>
          <div className="grid-item">{i}</div>
          <div className="grid-item">{row.nametype}</div>
          <div className="grid-item">{row.recclass}</div>
          <div className="grid-item">{row.mass}</div>
          <div className="grid-item">{row.fall}</div>
          <div className="grid-item">{row.year}</div>
          <div className="grid-item">{row.reclat}</div>
          <div className="grid-item">{row.reclong}</div>
        </Fragment>
      );
    }

    );
    return (
    <section className="grid center-align p-2">
      <ResultHeadings />
      {results}
    </section>
    );
  }
}