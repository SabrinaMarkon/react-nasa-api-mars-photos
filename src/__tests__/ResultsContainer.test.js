import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

// components
import ResultsContainer from '../containers/ResultsContainer';
import ResultHeadings from '../components/ResultHeadings';
import ResultRow from '../components/ResultRow';

function setup() {
  const wrapper = shallow(<ResultsContainer />);
  return wrapper;
}

describe('Testing ResultsContainer component', () => {
  it('renders the search results box', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a child ResultHeading component', () => {
    const wrapper = shallow(<ResultHeadings />);
    expect(wrapper.exists()).toBe(true);
  });

  it('should render a child ResultRow', () => {
    const wrapper = shallow(<ResultRow />);
    expect(wrapper.exists()).toBe(true);
  });
});