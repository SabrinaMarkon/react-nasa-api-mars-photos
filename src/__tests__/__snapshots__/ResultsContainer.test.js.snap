// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ResultsContainer component renders the search results box 1`] = `
<section>
  <ResultHeadings />
</section>
`;
